src/service_info.rs

```diff
@@ -499,6 +499,20 @@ impl TxtProperties {
     pub fn get_property_val_str(&self, key: &str) -> Option<&str> {
         self.get(key).map(|x| x.val_str())
     }
+
+    /// Returns a map of properties, where the key is the property key.
+    pub fn into_property_map_str(self) -> HashMap<String, String> {
+        self.properties
+            .into_iter()
+            .filter_map(|p| {
+                // Skip the property if the value is non-empty and not valid UTF-8.
+                let value = p
+                    .val
+                    .map_or(Some(String::new()), |v| String::from_utf8(v).ok())?;
```

Comment:
I would use [is_some_and](https://doc.rust-lang.org/std/option/enum.Option.html#method.is_some_and) instead of `map_or`, in this way we do not consider empty `String`s

Reply:
related to the previous comment resolution.